### PR TITLE
SDI-613 missing tests and tidy

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesService.kt [hunk 1]
```diff
@@ -106,7 +106,7 @@ class ActivitiesService(
       endDate, internalLocation?.id?.toLong(),
       pay.map { p ->
         PayRateRequest(
-          incentiveLevel = p.incentiveLevel!!, // TODO this should be made non-nullable in the Activities API soon
+          incentiveLevel = p.incentiveLevel!!, // TODO SDI-622 this should be made non-nullable in the Activities API soon
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesService.kt:109]: Attached JIRA tickets to all TODO to help us not forget them.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesServiceTest.kt [hunk 2]
```diff
@@ -32,13 +32,13 @@ import java.time.LocalDate
...
-private const val offenderProgramReferenceId: Long = 500
-private const val bookingId: Long = 600
-private const val offenderNo = "A1234AA"
+private const val ACTIVITY_SCHEDULE_ID: Long = 100
+private const val ACTIVITY_ID: Long = 200
+private const val NOMIS_COURSE_ACTIVITY_ID: Long = 300
+private const val ALLOCATION_ID: Long = 400
+private const val OFFENDER_PROGRAM_REFERENCE_ID: Long = 500
+private const val BOOKING_ID: Long = 600
+private const val OFFENDER_NO = "A1234AA"
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesServiceTest.kt:41]: Intellij warning on constant names.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivityToNomisIntTest.kt [hunk 3]
```diff
@@ -30,7 +30,7 @@ private const val COURSE_ACTIVITY_ID: Long = 300
 private const val ALLOCATION_ID: Long = 400
 private const val BOOKING_ID: Long = 500
 
-class ActivityToNomisTest : SqsIntegrationTestBase() {
+class ActivityToNomisIntTest : SqsIntegrationTestBase() {
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivityToNomisIntTest.kt:33]: Make integration tests easier to spot

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivityToNomisIntTest.kt [hunk 4]
```diff
@@ -166,20 +157,21 @@ class ActivityToNomisTest : SqsIntegrationTestBase() {
...
-            )
-          ).build()
-      ).get()
+      awsSnsClient.publish(amendActivityEvent()).get()
 
       await untilAsserted { assertThat(awsSqsActivityDlqClient!!.countAllMessagesOnQueue(activityDlqUrl!!).get()).isEqualTo(1) }
       nomisApi.verify(exactly(0), putRequestedFor(urlEqualTo("/activities/$COURSE_ACTIVITY_ID")))
     }
+
+    private fun amendActivityEvent(): PublishRequest? =
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivityToNomisIntTest.kt:166]: Remove duplciation to make tests read better

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/services/NomisApiServiceTest.kt [hunk 5]
```diff
@@ -247,6 +247,43 @@ internal class NomisApiServiceTest {
     }
   }
 
+  @Nested
+  inner class UpdateActivity {
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/services/NomisApiServiceTest.kt:251]: Add missing Nomis API tests

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**